src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -425,18 +435,62 @@ def rename_groups(self):
                 continue
             logger.info(f"Renaming: {migrated_group.name_in_workspace} -> {migrated_group.temporary_name}")
             tasks.append(
-                functools.partial(self._rename_group, migrated_group.id_in_workspace, migrated_group.temporary_name)
+                functools.partial(
+                    self._rename_group_and_wait,
+                    migrated_group.id_in_workspace,
+                    migrated_group.name_in_workspace,
+                    migrated_group.temporary_name,
+                )
             )
-        _, errors = Threads.gather("rename groups in the workspace", tasks)
-        if len(errors) > 0:
-            raise ManyError(errors)
+        renamed_groups = Threads.strict("rename groups in the workspace", tasks)
+        # Renaming is eventually consistent, and the tasks above have each polled to verify their rename completed.
+        # Here we also check that enumeration yields the updated names; this is necessary because otherwise downstream
+        # tasks (like reflect_account_groups_on_workspace()) may skip a renamed group because it doesn't appear to be
+        # present.
+        self._wait_for_renamed_groups(renamed_groups)
+
+    def _rename_group_and_wait(self, group_id: str, old_group_name, new_group_name: str) -> tuple[str, str]:
+        logger.debug(f"Renaming group {group_id}: {old_group_name} -> {new_group_name}")
```

Comment:
Maybe move this into the `_rename_group`

Reply:
I considered that, but it would move it within the scope of the retry so that upon retry you'd end up with repeated log entries stating the same thing. At this log level there's already logging from the retry handler (assuming that it's at the same log level) that shows what's going on.